isolates/bin/send_and_receive.dart

```diff
@@ -0,0 +1,44 @@
+// Read the file, spawn an isolate, send the file contents to the spawned
+// isolate, and wait for the parsed JSON.
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:isolate';
+
+final filename = 'json_01.json';
+
+Future<void> main() async {
+  // Read some data.
+  final fileData = await File(filename).readAsString();
+  final jsonData = await _spawnIsolateAndSendJson(fileData);
+
+  // Use that data.
+  print('Received JSON with ${jsonData.length} keys');
+}
+
+// Sends a JSON string to the spawned isolate to be parsed.
+Future<Map<String, dynamic>> _spawnIsolateAndSendJson(String jsonString) async {
```

Comment:
This is unnecessarily complicated when we know we only want a single computation.
Send the string as part of the original message, whether typed or untyped, either is fine.

So:
```dart
Future<Map<String, dynamic>> _spawnIsolateAndSendJson(String jsonString) async {
  final p = ReceivePort();
  await Isolate.spawn(_receiveAndParseJson, [p.sendPort, jsonString]);
  return (await p.first) as Map<String, dynamic>;
}

void _receiveAndParseJson(List<dynamic> args) {
  SendPort responsePort = args[0];
  String jsonString = args[1];
  var result = jsonDecode(jsonString)
  isolate.exit(responsePort, result);
}
```

Reply:
I'd suggest moving the reading of the file to the worker isolate, so something like this:

```dart
Future<void> main() async {
  // Read some data.
  final filename = 'json_01.json';
  final jsonData = await _spawnIsolateAndSendJson(filename);

  // Use that data.
  print('Received JSON with ${jsonData.length} keys');
}

Future<Map<String, dynamic>> _spawnIsolateAndSendJson(String fileName) async {
  final p = ReceivePort();
  await Isolate.spawn(_receiveAndParseJson, [p.sendPort, fileName]);
  return (await p.first) as Map<String, dynamic>;
}

void _receiveAndParseJson(List<dynamic> args) async {
  SendPort responsePort = args[0];
  String fileName = args[1];

  final fileData = await File(fileName).readAsString();
  var result = jsonDecode(fileData);
  Isolate.exit(responsePort, result);
}
```

Reply:
Oh, and if so, maybe rename `_receiveAndParseJson` to `_retrieveAndParseJson `

Reply:
Since it's no longer receiving events explicitly, maybe `_readAndParseJson`?

Reply:
Resolved in 2cbb16d